raiden/src/errors/transaction.rs

```diff
@@ -0,0 +1,162 @@
+use std::fmt;
+
+use thiserror::Error;
+
+const TRANSACTION_CANCELLED_MESSAGE_PREFIX: &str = "Transaction cancelled, please refer cancellation reasons for specific reasons";
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct RaidenTransactionCancellationReasons(pub Vec<Option<RaidenTransactionCancellationReason>>);
+
+impl RaidenTransactionCancellationReasons {
+    // If `message` is unexcepted format, [RaidenTransactionCancellationReason::Unknown] is returned instead of Err(_)
+    pub fn from_str(message: &str) -> Self {
+        if !message.starts_with(TRANSACTION_CANCELLED_MESSAGE_PREFIX) {
+            return RaidenTransactionCancellationReasons(vec![
+                Some(RaidenTransactionCancellationReason::Unknown)
+            ])
+        }
+
+        RaidenTransactionCancellationReasons(message[TRANSACTION_CANCELLED_MESSAGE_PREFIX.len()..]
+            .trim_matches(|c| char::is_whitespace(c) || c == '[' || c == ']')
+            .split(",")
+            .map(str::trim)
+            .map(|reason| {
+                match reason {
+                    "None" => None,
+                    reason => Some(RaidenTransactionCancellationReason::from_str(reason)),
+                }
+            })
+            .collect())
+    }
+
+    // as_error returns first error or Unknow
+    pub fn to_error(&self) -> RaidenTransactionCancellationReason {
```

Comment:
@proelbtn Thanks for your great work!!

Let me confirm, what usecase is this function used in?
So if dynamo return `[ThrottlingError, ConditionalCheckFailed]` as reasons, this function return `ThrottlingError`.
I am concerned that this may be confusing.

Reply:
In response to the discussion on Slack, we decided to delete this function. I deleted.